pkg/awsutils/awsutils.go

```diff
@@ -343,19 +358,61 @@ func (cache *EC2InstanceMetadataCache) initWithEC2Metadata() error {
 	}
 	log.Debugf("Found vpc-ipv4-cidr-block: %s ", cache.vpcIPv4CIDR)
 
-	// retrieve vpc-ipv4-cidr-blocks
+	// initSGIDs retrieves security groups
+	err = cache.initSGIDs(mac)
+	if err != nil {
+		return err
+	}
+
+	// initVPCIPv4CIDRs retrieves VPC IPv4 CIDR blocks
+	err = cache.initVPCIPv4CIDRs(mac)
+	if err != nil {
+		return err
+	}
+
+	// refresh security groups and VPC CIDR blocks in the background
+	go wait.Forever(func() { cache.initSGIDs(mac) }, 30*time.Second)
+	go wait.Forever(func() { cache.initVPCIPv4CIDRs(mac) }, 30*time.Second)
+
+	select {
+	case <-ctx.Done():
+		return nil
+	}
+}
+
+// initSGIDs retrieves security groups
+func (cache *EC2InstanceMetadataCache) initSGIDs(mac string) error {
```

Comment:
Instead of `initXXX` can we name these functions `refreshXXX` to better indicate that they are not just run on initialization of the cache?

Reply:
> Instead of `initXXX` can we name these functions `refreshXXX` to better indicate that they are not just run on initialization of the cache?

I like the idea but we call these functions once during init and then spawn a go-routine. Instead, how does  something more generic like `fetchSGIDs` sound ?